File: CivitasJava/src/Civitas/CasillaCarcel.java
package Civitas;

import java.util.ArrayList;

public class CasillaCarcel extends Casilla{
    private int carcel;
    
    CasillaCarcel(String nombre,int numCasillaCarcel){
        super(nombre);
        this.carcel = numCasillaCarcel;
    }

    @Override    
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {
        if(super.jugadorCorrecto(iactual,todos)){
            super.informe(iactual,todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }
    
}

File: CivitasRuby/lib/casilla_carcel.rb
# encoding:utf-8
require_relative 'casilla'
module Civitas
  class CasillaCarcel < Casilla
    def initialize(nombre,num_casilla_carcel)
      super(nombre)
      @carcel = num_casilla_carcel
    end

    def recibe_jugador(iactual,todos)
      if jugador_correcto(iactual,todos)
          informe(iactual,todos)
          todos[iactual].encarcelar(@carcel)
      end
    end
  end
end
